File: How To Find Person/Ruby/WhitePages-PersonLookup/app/models/location.rb
class Location

  attr_accessor :entity, :city, :postal_code, :state_code, :country_code,
                :is_receiving_mail, :usage, :delivery_point, :is_deliverable, :standard_address_line1, :standard_address_line2

  def initialize(entity)
    @entity = entity
    @city = city
    @postal_code = postal_code
    @state_code = state_code
    @country_code = country_code
    @is_receiving_mail = is_receiving_mail
    @usage = usage
    @delivery_point = delivery_point
    @is_deliverable = is_deliverable
    @standard_address_line1 = standard_address_line1
    @standard_address_line2 = standard_address_line2
  end

  def data
    {
     :city => city,
     :postal_code => postal_code,
     :state_code => state_code,
     :country_code => country_code,
     :is_receiving_mail => is_receiving_mail,
     :usage => usage,
     :delivery_point => delivery_point,
     :is_deliverable => is_deliverable,
     :standard_address_line1 => standard_address_line1,
     :standard_address_line2 => standard_address_line2
    }
  end

  private

  def city
    entity['city']
  end

  def postal_code
    entity['postal_code']
  end

  def state_code
    entity['state_code']
  end

  def country_code
    entity['country_code']
  end

  def is_receiving_mail
    entity['is_receiving_mail']
  end

  def is_deliverable
    entity['is_deliverable']
  end

  def delivery_point
    entity['delivery_point']
  end

  def usage
    entity['usage']
  end

  def standard_address_line1
    entity['standard_address_line1']
  end

  def standard_address_line2
    entity['standard_address_line2']
  end
end

File: How To Find Person/PHP/WhitePages-PersonLookup/lib/location.php
<?php
/**
 * Library for abstraction of methods to use Whitepages Pro API
 * @require     PHP5
 *
 * @author      Kushal Shah
 * @date        2014-06-02
 */

class Location
{
    public $entity;
    public $city;
    public $postal_code;
    public $state_code;
    public $country_code;
    public $is_receiving_mail;
    public $usage;
    public $delivery_point;
    public $is_deliverable;
    public $standard_address_line1;
    public $standard_address_line2;


    public function __construct($entity)
    {
        $this->entity = $entity;
        $this->city = $this->city();
        $this->postal_code = $this->postal_code();
        $this->state_code = $this->state_code();
        $this->country_code = $this->country_code();
        $this->is_receiving_mail = $this->is_receiving_mail();
        $this->usage = $this->usage();
        $this->delivery_point = $this->delivery_point();
        $this->is_deliverable = $this->is_deliverable();
        $this->standard_address_line1 = $this->standard_address_line1();
        $this->standard_address_line2 = $this->standard_address_line2();
    }

    public function data()
    {
        return array(
            'city' => $this->city,
            'postal_code' => $this->postal_code,
            'state_code' => $this->state_code,
            'country_code' => $this->country_code,
            'is_receiving_mail' => $this->is_receiving_mail,
            'usage' => $this->usage,
            'delivery_point' => $this->delivery_point,
            'is_deliverable' => $this->is_deliverable,
            'standard_address_line1' => $this->standard_address_line1,
            'standard_address_line2' => $this->standard_address_line2
        );
    }


    private function city()
    {
        return  $this->entity['city'];
    }

    private function postal_code()
    {
        return  $this->entity['postal_code'];
    }

    private function state_code()
    {
        return  $this->entity['state_code'];
    }

    private function country_code()
    {
        return  $this->entity['country_code'];
    }

    private function is_receiving_mail()
    {
        return  $this->entity['is_receiving_mail'];
    }

    private function usage()
    {
        return  $this->entity['usage'];
    }

    private function delivery_point()
    {
        return  $this->entity['delivery_point'];
    }

    private function is_deliverable()
    {
        return  $this->entity['is_deliverable'];
    }

    private function standard_address_line1()
    {
        return  $this->entity['standard_address_line1'];
    }

    private function standard_address_line2()
    {
        return  $this->entity['standard_address_line2'];
    }
}

